### PR TITLE
fix(orchestrator): require review for new runtime tools

### LINKED ISSUE #1787: Security: add security review gate for new runtime tools
## Description
Implement an LLM-friendly, independently shippable backlog item for **add security review gate for new runtime tools**. This issue should stay narrow enough for one agent/PR while producing durable behavior, tests, and documentation where appropriate.

## Affected locations
- Approval/governor/security policy surfaces
- Runtime state, logs, prompts, and delivery integrations
- Agent tool wrappers and web/control endpoints

## Recommendation
Add the smallest safe control that addresses this risk, prefer deny-by-default behavior for unsafe paths, document operator override semantics, and include regression tests for both allowed and denied flows.

## Goals
Deliver add security review gate for new runtime tools as a discrete capability or hardening change.
Make the behavior deterministic enough for autonomous workers to verify.
Keep the implementation small enough for a one-issue worker and one PR.
Capture operator/user-facing behavior in docs or help text when applicable.

## Expected outcomes
add security review gate for new runtime tools is available, documented, and test-backed.
Future PM/worker handoffs can point at this issue without needing extra interpretation.
Failure modes are explicit and produce actionable messages instead of silent drift.
The change does not broaden into adjacent backlog work or unrelated refactors.

## Suggested implementation slices
Inventory the current code/docs paths that own this behavior.
Add or update the narrow model/config/schema needed for the feature or guardrail.
Implement the smallest runtime/docs change that satisfies the acceptance criteria.
Add focused fixtures/tests for success, failure, and edge cases.
Update docs/help output and handoff notes so later agents can use it.

## Acceptance criteria
- [ ] There is a dedicated implementation or documentation change for add security review gate for new runtime tools.
- [ ] The control defaults to the safest practical behavior and requires explicit override for risky cases.
- [ ] At least one automated test, fixture, or deterministic verifier covers the intended behavior.
- [ ] Sensitive payloads/secrets are not printed in logs, test output, issue comments, or telemetry.
- [ ] At least one negative/edge case is covered so the failure mode is explicit.
- [ ] Docs/help text or inline operator guidance explains how to use or interpret the behavior.
- [ ] The PR description links this issue and summarizes verification evidence.
- [ ] No unrelated production behavior is changed outside the stated scope.

## LLM handoff notes
- Treat this as a one-issue/one-PR unit.
- Start by searching for existing implementations and tests before adding new surfaces.
- Prefer structured outputs and fixtures over prose-only behavior.
- If this overlaps an existing open issue, comment with evidence and stop for PM deduplication instead of creating an unscoped implementation.
- Do not perform broad refactors or change unrelated worker/PM behavior while implementing this ticket.

## Source
Created by the 2026-07-11 fourth strategic backlog expansion run requesting 100 issues. Topic: **Security**.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- Default installed runtime tool manifests to require HITL security review unless explicitly reviewed safe with `requiresHitl: false`
- Treat MCP skills without `tools.json` as approval-required because their concrete runtime tools are unknown
- Document the runtime skill-tool review semantics in the quickstart

## Test Plan
- `npm --workspace @franken/orchestrator test -- tests/unit/skills/skill-manager.test.ts tests/unit/adapters/mcp-execution-adapters.test.ts tests/unit/skills/provider-skill-translator.test.ts`
- `npm run build --workspace @franken/types && npm run build --workspace @franken/planner && npm run build --workspace @franken/brain && npm run build --workspace @franken/observer && npm run build --workspace @franken/critique && npm run build --workspace @franken/governor && npm --workspace @franken/orchestrator run typecheck`
- `npm --workspace @franken/orchestrator run build`
- `npm --workspace @franken/orchestrator run lint` (passes with pre-existing warnings)

Closes #1787

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/skills/skill-manager.ts [hunk 1]
```diff
@@ -81,9 +90,12 @@ export class SkillManager {
     );
 
     if (catalogEntry.toolDefinitions?.length) {
+      const tools = requireSecurityReviewByDefault(
+        SkillToolManifestSchema.parse(catalogEntry.toolDefinitions),
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/skills/skill-manager.ts]: Fixed in af67f209. SkillManager.install now validates and defaults catalog toolDefinitions before creating the skill directory or writing mcp.json, and the unit suite covers invalid tool definitions leaving no partial install.

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/skills/skill-manager.ts [hunk 2]
```diff
@@ -80,10 +94,10 @@ export class SkillManager {
       JSON.stringify(mcpConfig, null, 2),
     );
 
-    if (catalogEntry.toolDefinitions?.length) {
+    if (tools) {
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/skills/skill-manager.ts:98]: Fixed in 7e65a7b0. Catalog reinstall without current toolDefinitions now removes an existing tools.json, and the regression suite verifies stale safe-tool manifests are not retained after reinstalling a skill with an unknown/new MCP server.

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/skills/skill-manager.ts [hunk 3]
```diff
@@ -80,11 +94,14 @@ export class SkillManager {
...
+    const toolsPath = join(skillDir, 'tools.json');
+    if (tools) {
       writeFileSync(
-        join(skillDir, 'tools.json'),
-        JSON.stringify(catalogEntry.toolDefinitions, null, 2),
+        toolsPath,
+        JSON.stringify(tools, null, 2),
       );
+    } else if (existsSync(toolsPath)) {
+      rmSync(toolsPath);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/skills/skill-manager.ts:104]: Fixed in 6e95b080. installCustom now removes any stale tools.json when replacing a skill with a custom/no-manifest MCP server, and regression coverage verifies a prior reviewed requiresHitl:false manifest is cleared.

**Comment by djm204** [comment]
@codex review
